### ISSUE TITLE
Line No showGaps with no data at start 

### ISSUE THREAD

**Issue by nonmetalhail** [PredixDev/px-vis#82]
If `showGaps = false` and the gap is in the beginning, the line draws to origin. Should make it gap until first data point. Check if similar behavior when gap is at end.

**Comment by davidsteinberger** [comment]
I just realized that too. Is that going to make it into v1.0.0?

**Comment by nonmetalhail** [comment]
Unfortunately no, it will have to be a patch after the release. 

The bug is introduced because I added options for how the chart handles missing data points. In the pre-v1 version, it would show a gap whenever one series did not have a datapoint at a particular timestamp/x coordinate. The new default behavior is that it spans those gaps to provide continuous lines.

You can still revert to the pre-v1 gap behavior by adding the attribute `show-gaps` to your chart. Hopefully this will work for you until it can be patched (or it might be the behavior you actually want).

As further explanation as to the differences and why this change was made and the difference between the two modes:

```
[
    {
        "series1": 3,
        "series2": 4,
        "x": 0
    }, {
        "series1": 1,
        "series2": 1,
        "x": 1
    }, {
        "series1": 3,
        "x": 2
    }, {
        "series1": 2,
        "series2": 4,
        "x": 3
    }, {
        "series1": 1,
        "series2": 2,
        "x": 4
    }
]
```

The above dataset is missing a `series2` value at the third `x` reading. With `show-gaps` in v1.x and in pre-v1 charts, this resulted in a continuous `series1` and a broken `series2` line. In the new default v1.x, this will show both series as continuous lines.

The reason for changing this behavior is that your series may have different sampling rates, say `series1` was sampled every minute while `series2` was sampled every 2 minutes. In this case, you wouldnt want to show `series2` broken. 

If you have a different use case and want gaps, then you have two options. First is to use `show-gaps` which will show a gap whenever a datapoint is `null` or missing. The second option is to leave `showGaps` as `false` but insert `null` values:

```
    {
        "series1": 3,
        "series2": null,
        "x": 2
    }
```

`Null` values will still provide a gap as the `showGaps` property only affects how the chart handles missing datapoints.

Despite all this, there is a bug when it tries to span a gap in the beginning of the series, so it tries to span from the origin point. 

Probably more info than you needed, but hopefully useful to others wondering what the difference is.

**Comment by davidsteinberger** [comment]
Definitely helpful! I came to the same conclusion when going through the code but missed the _null_ options. Thanks for pointing that out. I went with _showGaps_ and it works a treat.

**Comment by nonmetalhail** [comment]
Fixed in v1.1
